api/data_test.go

```diff
@@ -65,6 +66,47 @@ func TestGetCacheTimeEndpoint(t *testing.T) {
 	})
 }
 
+func TestGetCacheTimeReturnsError404(t *testing.T) {
+	Convey("Given a GetCacheTime handler", t, func() {
+		ctx := context.Background()
+		mockedDataStore := &mock.DataStoreMock{
+			GetCacheTimeFunc: func(ctx context.Context, id string) (*models.CacheTime, error) {
+				return nil, errs.ErrCacheTimeNotFound
+			},
+		}
+		dataStoreAPI := setupAPIWithStore(ctx, mockedDataStore)
+		Convey("When a non-existent cache time is requested with its ID", func() {
+			var nonExistentCacheID = "abcdef0a1b2c3d4e5f67890123456789"
+			r := httptest.NewRequest(http.MethodGet, baseURL+nonExistentCacheID, http.NoBody)
+			w := httptest.NewRecorder()
+			dataStoreAPI.Router.ServeHTTP(w, r)
+			Convey("Then unmatched cache time is not found with status code 404", func() {
+				So(w.Code, ShouldEqual, http.StatusNotFound)
+			})
+		})
+	})
+}
+
+func TestGetCacheTimeReturnsError500(t *testing.T) {
+	Convey("Given a GetCacheTime handler", t, func() {
+		ctx := context.Background()
+		mockedDataStore := &mock.DataStoreMock{
+			GetCacheTimeFunc: func(ctx context.Context, id string) (*models.CacheTime, error) {
+				return nil, errs.ErrDataStore
+			},
+		}
+		dataStoreAPI := setupAPIWithStore(ctx, mockedDataStore)
+		Convey("When the api cannot connect to datastore", func() {
```

Comment:
That's not really true, what you're really testing is the scenario in which there is an error with the data store. That error does not necessarily have to be a connection error.

Reply:
updated it